### PR TITLE
Update osmosis cw-staking adapter to query the pool id via ans pool

### PR DESCRIPTION
**Pull request by adairrr**
# Checklist

- CI is green.
- Changelog updated.

### CODE REVIEW

File: integrations/osmosis/packages/abstract-adapter/src/lib.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
 pub const OSMOSIS: &str = "osmosis";
-pub const AVAILABLE_CHAINS: &[&str] = &["osmosis"];
+pub const AVAILABLE_CHAINS: &[&str] = &["osmosis", "osmo-test"];
```
Comment (Buckram123) [inline integrations/osmosis/packages/abstract-adapter/src/lib.rs:2]: :rocket:

**Review by Buckram123 — APPROVED** [review]
Looks good, would love to see that in action! Do you think we should cover this new feature by tests?

File: integrations/osmosis/packages/abstract-adapter/src/lib.rs (hunk 1 above)
Comment (adairrr) [reply integrations/osmosis/packages/abstract-adapter/src/lib.rs:2]: And we should probably use `abstract_sdk::core::registry::OSMOSIS` here, as long as we include `osmo-test` there.

**Comment by adairrr** [comment]
> Looks good, would love to see that in action! Do you think we should cover this new feature by tests?

Yes! We should definitely be testing this even if it's just some unit tests with mocked ans data. 

We also should likely need to add some better error messages for when the pool isn't found or just otherwise to clarify the stack trace.

**Comment by Buckram123** [comment]
Update: we need a way to replace LockupQuerier, because it seems like any `/osmosis.lockup.Query/*` is not allowed for contracts on osmosis :cry:

**Comment by CyberHoward** [comment]
@Buckram123 `/osmosis.lockup.Query/*` is now enabled, could we move forward with finishing this pr?

**Comment by CyberHoward** [comment]
@Buckram123 is this still relevant?

**Comment by Buckram123** [comment]
Merged in #30

**Pull request closed without merging**